lib/sinatra/base.rb

```diff
@@ -842,6 +842,7 @@ def compile_template(engine, data, options, views)
           body, path, line = settings.templates[data]
           if body
             body = body.call if body.respond_to?(:call)
+            body = body.dup if body.frozen?
```

Comment:
Can you provide a case where body is actually immutable from here? Maybe an empty response?